libraries/esp8266/examples/I2STransmit/I2STransmit.ino

```diff
@@ -0,0 +1,75 @@
+/*
+   I2S stereo microphone (input) UDP transmitter
+   Needs a UDP listener (netcat/etc.) on port 8266 on the PC
+
+   Under Linux:
+    nc -u -p 8266 -l | play -t raw -r 11025 -b 16 -c 2 -e signed-integer -
+
+   Released to the Public Domain by Earle F. Philhower, III
+*/
+
+#include <ESP8266WiFi.h>
+#include <WiFiUdp.h>
+#include <i2s.h>
+
+// Set your network here
+const char *SSID = "....";
+const char *PASS = "....";
+
+WiFiUDP udp;
+// Set your listener PC's IP here:
+const IPAddress listener = { 192, 168, 1, 2 };
+const int port = 8266;
+
+int16_t buffer[100][2]; // Temp staging for samples
+
+void setup() {
+  Serial.begin(115200);
+
+  // Connect to WiFi network
+  Serial.println();
+  Serial.println();
+  Serial.print("Connecting to ");
+  Serial.println(SSID);
+
+  WiFi.begin(SSID, PASS);
+
+  while (WiFi.status() != WL_CONNECTED) {
+    delay(500);
+    Serial.print(".");
+  }
+  Serial.println("");
+  Serial.println("WiFi connected");
+  Serial.print("My IP: ");
+  Serial.println(WiFi.localIP());
+
+  i2s_rxtx_begin(true, false); // Enable I2S RX
+  i2s_set_rate(11025);
+
+  Serial.print("\nStart the listener on ");
+  Serial.print(listener);
+  Serial.print(":");
+  Serial.println(port);
+  Serial.println("ex:  nc -u -p 8266 -l | play -t raw -r 11025 -b 16 -c 2 -e signed-integer -");
+
+  udp.beginPacket(listener, port);
+  udp.write("I2S Receiver\r\n");
+  udp.endPacket();
+
+}
+
+void loop() {
+  static int cnt = 0;
+  // Each loop will send 100 raw samples (400 bytes)
+  // UDP needs to be < TCP_MSS which can be 500 bytes in LWIP2
```

Comment:
I come after the battle with saying that I'm not sure MSS applies to UDP.
UDP is only limited by MTU which is always 1460 on ESP8266 (defined and given by PHY).

Reply:
Are you sure about that?  I started with ~1024 bytes per packet, but no UDP packets ever made it to the PC with the low-mem LWIP.  At high-mem it did work.  Dropping to 400 bytes or so made things work under both so I left it there.  It's also better for I2S latency, since you want to minimize the amount of buffering, anyway.

Reply:
Yes, I'm sure that it should work, and I'm now sure I have to check why it does not :)